rclpy/rclpy/task.py

```diff
@@ -114,12 +125,17 @@ def set_result(self, result: T) -> None:
         """
         Set the result returned by a task.
 
+        :raises: RuntimeError if future state is not pending
+
         :param result: The output of a long running task.
         """
         with self._lock:
+            if not self._pending():
+                raise RuntimeError(f"Cannot set result from state: {self._state}")
+            
```

Comment:
this breaks the following test (possibly userspace):

https://github.com/ros2/rclpy/blob/296425c4183135c048dff8fd22a4c0d44b39b0f4/rclpy/test/test_rosout_subscription.py#L114-L115